src/ape/utils.py

```diff
@@ -513,45 +557,106 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
+    NOTE: do not appear in a Pydantic model's set of properties
+    """
+
+    def __get__(self, *args):
+        raise ValueError("Value not set. Please inject this property before calling.")
+
+
+class ManagerAccessBase:
+
+    # NOTE: cast is used to update the class type returned to mypy
+    account_manager: ClassVar["AccountManager"] = cast("AccountManager", injected_before_use())
+
+    chain_manager: ClassVar["ChainManager"] = cast("ChainManager", injected_before_use())
+
+    compiler_manager: ClassVar["CompilerManager"] = cast("CompilerManager", injected_before_use())
     """
-    A `data class <https://docs.python.org/3/library/dataclasses.html>`__ that
-    is also abstract (meaning it has methods that **must** be implemented in a
-    sub-class or else errors will occur). This class cannot be instantiated
-    on its own.
+    The group of compiler plugins for compiling source files. See
+    :class:`~ape.managers.compilers.CompilerManager` for more information.
+    Call method :meth:`~ape.managers.project.ProjectManager.load_contracts` in this class
+    to more easily compile sources.
     """
 
+    config_manager: ClassVar["ConfigManager"] = cast("ConfigManager", injected_before_use())
+    """
+    A reference to :class:`~ape.managers.config.ConfigManager`, which
+    manages project and plugin configurations.
+    """
 
-abstractdataclass = partial(dataclass, kwargs=True, meta=AbstractDataClassMeta)
-"""
-A `data class <https://docs.python.org/3/library/dataclasses.html>`__ that is
-also abstract (meaning it has methods that **must** be implemented or else
-errors will occur. This class cannot be instantiated on its own.
-"""
+    conversion_manager: ClassVar["ConversionManager"] = cast(
+        "ConversionManager", injected_before_use()
+    )
+    """
+    A reference to the conversion utilities in
+    :class:`~ape.managers.converters.ConversionManager`.
+    """
 
+    dependency_manager: ClassVar["_DependencyManager"] = cast(
+        "_DependencyManager", injected_before_use()
+    )
 
-class injected_before_use(property):
+    network_manager: ClassVar["NetworkManager"] = cast("NetworkManager", injected_before_use())
     """
-    Injected properties are injected class variables that must be set before use
-    NOTE: do not appear in a Pydantic model's set of properties
+    The manager of networks, :class:`~ape.managers.networks.NetworkManager`.
+    To get the active provide, use
+    :py:attr:`ape.managers.networks.NetworkManager.active_provider`.
     """
 
-    def __get__(self, *args):
-        raise ValueError("Value not set. Please inject this property before calling.")
+    plugin_manager: ClassVar["PluginManager"] = cast("PluginManager", injected_before_use())
+
+    project_manager: ClassVar["ProjectManager"] = cast("ProjectManager", injected_before_use())
+
+    query_manager: ClassVar["QueryManager"] = cast("QueryManager", injected_before_use())
+
+
+class AbstractBase(ManagerAccessBase, ABC):
```

Comment:
This is sort've why I used `API` instead of `Abstract` or `Base` everywhere

`BaseAPI` would look nicer here

Reply:
My question for us is this:  is "API" a marketing term for our abstract plugin classes?

We should avoid terms that are likely to change in the future. "Astract" is general enough that I don't see it ever changing. However, if we are confident in the same regarding "API", then I like that too.

My only point is to avoid the term if and only if it is ape-specific and not a general term for the idea. I know for sure "abstract" is general enough, but "API" can often refer to the actual implementation of an interface which may confuse users who are not familiar with ape-isms.

Reply:
My question for us is this:  is "API" a marketing term for our abstract plugin classes?

We should avoid terms that are likely to change in the future. "Astract" is general enough that I don't see it ever changing. However, if we are confident in the same regarding "API", then I like that too.

My only point is to avoid the term if and only if it is ape-specific and not a general term for the idea. I know for sure "abstract" is general enough, but "API" can often refer to the actual implementation of an interface which may confuse users who are not familiar with ape-isms.

Reply:
How about just `BaseInterface` (`AccountInterface`, `ProviderInterface`, etc.)

Reply:
Note: bikeshed naming in another PR